src/telliot_feeds/sources/gas_price_oracle.py

```diff
@@ -2,127 +2,125 @@
+            return right_block
+        if right_block["timestamp"] > target_timestamp:
+            return left_block
+
+        if left_block["timestamp"] - target_timestamp <= right_block["timestamp"] - target_timestamp:
+            return left_block
         else:
-            logger.warning("Can't fetch data: GasPriceOracle DataSource QueryParameters unset")
-            return None
+            return right_block
 
-    async def fetch_new_datapoint(
-        self,
-    ) -> OptionalDataPoint[Any]:
-        """Retrieves historical gas prices from Owlracle API.
+    async def fetch_new_datapoint(self) -> OptionalDataPoint[Any]:
+        """Fetch median gas price for a given timestamp by fetching
+        block info for the closest block.
 
         Returns:
-            float gas price in gwei, typically with one decimal place
+            Current time-stamped value
         """
-        rsp = await self.fetch_historical_gas_price()
-        if rsp is None:
-            logger.warning("No response from GasPriceOracle API")
-            return None, None
-
-        if rsp.status_code // 100 != 2:
-            logger.warning("Invalid response from GasPriceOracle API: " + str(rsp.status_code))
-            return None, None
-
+        if not self.chainId:
+            raise ValueError("Chain ID not provided")
         try:
-            historical_gas_prices = rsp.json()
-        except JSONDecodeError as e:
-            logger.error("GasPriceOracle source returned invalid JSON:", e.strerror)
-            return None, None
-
-        if historical_gas_prices == []:
-            logger.warning("GasPriceOracle source returned no historical gas prices.")
+            self.web3 = update_web3(self.chainId, self.cfg)
+        except Exception as e:
+            logger.warning(f"Error occurred while updating web3 instance: {e}")
             return None, None
 
-        gas_prices = []
+        if not self.web3:
+            raise ValueError("Web3 not instantiated")
 
+        nearest_block = self.binary_search_block()
+        if nearest_block is None:
+            return None, None
         try:
-            for i in historical_gas_prices:
-                # find avg of high and low
-                avg = (i["gasPrice"]["high"] + i["gasPrice"]["low"]) / 2
-
-                gas_prices.append(avg)
-        except KeyError:
-            logger.error("Unable to parse GasPriceOracle source JSON")
+            block = self.web3.eth.get_block(nearest_block["number"], full_transactions=True)
+        except Exception as e:
+            logger.error(f"Error occurred while fetching block: {e}")
             return None, None
-        except ValueError:
-            logger.error("Unable to calculate median gas price from GasPriceOracle source JSON")
+        # sort the transactions by gas price
+        try:
+            sorted_gas_prices = sorted(tx["gasPrice"] for tx in block["transactions"])  # type: ignore
```

Comment:
Why's the typing ignored here? And is a try except needed? Can you avoid doing so if you just omit `None` or ones that are missing the required key like so:
```py
sorted_gas_prices = sorted(tx["gasPrice"] for tx in block["transactions"] if tx.get("gasPrice") is not None)
```
?

Reply:
Because the block.transactions is a Union[txdata, HexBytes] which returns one or the other depending on fullTransaction or not.  will refactor to get rid of the ignore